### PR TITLE
[TIMOB-23995] iOS: Add __dirname and __filename to the global app.js scope

### PR DESCRIPTION
**Pull request by hansemannn**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-23995

Extended test-case to validate files `require()`'d can be found in the linked JIRA.

### CODE REVIEW

File: iphone/Classes/KrollBridge.m [hunk 1]
```diff
@@ -555,6 +555,16 @@ - (void)didStartNewContext:(KrollContext *)kroll
...
   TiStringRef globalPropertyName = TiStringCreateWithCFString((CFStringRef) @"global");
   TiObjectSetProperty(jsContext, globalRef, globalPropertyName, globalRef, kTiPropertyAttributeDontEnum | kTiPropertyAttributeReadOnly | kTiPropertyAttributeDontDelete, NULL);
+  TiStringRelease(globalPropertyName);
+
+  // Set the __dirname and __filename for the app.js.
+  // For other files, it will be injected via the `TitaniumModuleRequireFormat` property
+  TiStringRef dirnameProperty = TiStringCreateWithCFString((CFStringRef) @"__dirname");
+  TiStringRef filenameProperty = TiStringCreateWithCFString((CFStringRef) @"__filename");
+  TiObjectSetProperty(jsContext, globalRef, dirnameProperty, [KrollObject toValue:kroll value:@"/"], kTiPropertyAttributeDontEnum | kTiPropertyAttributeReadOnly | kTiPropertyAttributeDontDelete, NULL);
+  TiObjectSetProperty(jsContext, globalRef, filenameProperty, [KrollObject toValue:kroll value:@"/app.js"], kTiPropertyAttributeDontEnum | kTiPropertyAttributeReadOnly | kTiPropertyAttributeDontDelete, NULL);
```
Comment (cb1kenobi) [inline iphone/Classes/KrollBridge.m]: `__filename` should just be the filename, no `/`.

**Review by cb1kenobi — CHANGES_REQUESTED** [review]

File: iphone/Classes/KrollBridge.m (hunk 1 above)
Comment (hansemannn) [reply iphone/Classes/KrollBridge.m]: It's with parity of the current implementation (for require'ish format). Then we should change it for both.

**Review by cb1kenobi — APPROVED** [review]
@hansemannn I'm sorry, I'm horribly mistaken. `__filename` is a full path. My bad.

**Review by janvennemann — APPROVED** [review]
Looks good!

**Pull request merged**